modules/commons/src/main/java/org/apache/synapse/commons/emulator/http/consumer/HttpResponseProcessor.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * *
+ *  * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied.  See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.apache.synapse.commons.emulator.http.consumer;
+
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.*;
+import io.netty.util.CharsetUtil;
+import org.apache.synapse.commons.emulator.http.dsl.dto.Header;
+import org.apache.synapse.commons.emulator.http.dsl.dto.consumer.IncomingMessage;
+import org.apache.synapse.commons.emulator.http.dsl.dto.consumer.OutgoingMessage;
+import org.apache.synapse.commons.emulator.http.dsl.HttpConsumerContext;
+import org.apache.synapse.commons.emulator.http.dsl.dto.Cookie;
+
+import java.util.Map;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.NOT_FOUND;
+
+
+public class HttpResponseProcessor {
+    private HttpConsumerContext consumerContext;
+
+    public HttpResponseProcessor(HttpConsumerContext consumerContext) {
+        this.consumerContext = consumerContext;
+    }
+
+    public void process(HttpRequestContext requestContext, ChannelHandlerContext ctx) {
+        OutgoingMessage outgoing = getMatchResource(requestContext);
+        if (outgoing == null) {
+            if (!write404NotFoundResponse(requestContext, ctx)) {
+                ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
+            }
+        } else {
+            if (!writeResponse(requestContext, outgoing, ctx)) {
+                ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
+            }
+        }
+    }
+
+    private boolean writeResponse(HttpRequestContext requestContext, OutgoingMessage outgoing, ChannelHandlerContext
```

Comment:
Line character count limit is 120 in a source file.

Reply:
Done